lib/cosmos/tools/packet_viewer/packet_viewer.rb

```diff
@@ -531,6 +535,7 @@ def self.run(option_parser = nil, options = nil)
             end
             options.packet = split
           end
+          option_parser.on("-r", "--rate PERIOD", "Set the polling rate to PERIOD (unit seconds)") { |arg| options.rate = arg.to_f } 
```

Comment:
This could very easily end up with a period of 0.0 which wouldn't be ideal.  Can you change the .to_f call to use Float() instead.

arg.to_f becomes Float(arg)

Thanks!